src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
```

Comment:
returning `null` in case of parse failure looks fine to me: this is user input, ie potentially garbage. An exception would force a mandatory try/catch around any call to getDate (which most will forget). Doing `if ($date = $foo->getDate(..))` has less boilerplate to me.

If we accept DateTimeInterface as default, then the return type hint should be changed to DateTimeInterface|null